### ISSUE TITLE
Replace Open Discussions Elasticsearch with OpenSearch

### ISSUE THREAD

**Issue by pdpinch** [mitodl/open-discussions#3494]
Elasticsearch 6.5 has passed it's end of life. Instead of upgrading Elasticsearch, we would like to switch to a hosted OpenSearch service.

#### Plan:

- [x] investigate what changes are necessary to switch from Elasticsearch to OpenSearch
- [x] create a branch for OpenSearch testing 
- [x] upgrade from 6.3 -> 6.8
- [x] Upgrade client from 6.8 -> 7.10
    - [x] Upgrade python packages
    - [x] Test local
    - [x] Merge and test RC
    - [x] Merge and test production
- [x] Upgrade service from 6.8 -> 7.10
    - [x] Update docker-compose settings and test locally
    - [x] Once PR is approved, will merge which won't affect RC at all
    - [x] Mike will update RC, we test RC
    - [x] Once RC is proven good, update prod
- [x] Swap OpenSearch client in for Elasticsearch client
    - [x] Update python packages & test local. PR
    - [x] Upgrade on RC
    - [x] Test - this is OS client with ES service
- [x] Swap to OpenSearch service
  - [x] Change docker-compose settings and test locally
  - [x] Once PR is approved, will merge which won't affect RC at all
  - [x] Mike will update RC, we test RC
  - [x] Once RC is proven good, update prod

#### AC:

- [x] Get Production client to OpenSearch 1.0.0
- [x] Get production service to OpenSearcch 1.0.0
- [x] Pass all tests
- [x] All previous search endpoints work
  - [x] Results do not break for OCW
  - [x] Results do not break for /learn/search
  - [x] Results do not break for posts/open discussions
  - [x] Suggested results work

**Comment by pdpinch** [comment]
@asadiqbal08 has done the research for this for micromasters and opened a PR for the upgrade, https://github.com/mitodl/micromasters/pull/5166. I'm not sure if he found any issues that would affect this project.

**Comment by JenniWhitman** [comment]
The deploy of the es7 client to RC has shown that there are compatibility issues between the es7 client and 6 service in regards to mapping structure that were not seen in prior testing. I have a plan for moving forward:
1. I'm pulling back the 7.10 branch and modifying it with a local 6.8 client and the changes needed for type. (this morning)
2. We'll test this branch locally and push again to RC (Today/Monday)
3. Mike can then test the upgrade to 7.10 service once the 7.10 client and 6.8 service are stable. (Early next week)
4. We then upgrade the service for local users in a separate PR (This PR is small & already tested as it's the other half of the 7.10 PR - just a change to the docker image)
5. The OpenSearch client PR is then applied. (Already ready)
6. The OpenSearch service will be updated and the local service will be updated in a separate PR. (again, small change to the docker image.

**Comment by JenniWhitman** [comment]
Nothing, Mike swapped production service to OS as well. All have been replaced.
